Remove Loading Spinner in favor of NProgress Bar

SudharakaP opened the pull request:
Related to https://github.com/opencollective/opencollective-frontend/pull/7904

https://user-images.githubusercontent.com/12435965/174993626-5815fe13-7598-4d3c-9701-6c7d792dd864.mp4

SudharakaP commented:
@znarf : Here, I've done a refactor to the `SearchForm.js` to show the spinner animation for around 3 seconds (the average time that takes to do a search in prod) when performing a search. This will ensure we have a consistent experience across all search forms. For example the search modal on the home page as well as the search form in the `/search` page both have this loading spinner applied by default. 

It would be nice to find a way to sync this spinner with the `NProgress` bar (or use the `NProgress` spinner) but I couldn't find a way to do so for the moment without making this too complicated. I think this is an acceptable solution given we also show the `NProgress` bar in all situations and although we don't have the exact sync between the `NProgress` bar and the spinner we the user has sufficient indication that the page is loading. 🤔

znarf commented on components/SearchForm.js:
```diff
@@ -49,15 +49,26 @@ class SearchForm extends React.Component {
-    const searchInput = event.target.elements.q;
-    this.setState({ isLoading: true });
-    this.props.router.push({ pathname: '/search', query: { q: searchInput.value } });
+    /*
+     * Here we set the loading state for a few seconds to show
+     * the loading spinner in place of the search button within the search bar.
+     */
+    this.setState({ isLoading: true }, () => {
+      setTimeout(() => {
+        this.setState({ isLoading: false });
+      }, 3000);
+    });
```
What happens after 3 seconds ?

SudharakaP replied:
After three seconds the button reverts back to normal state. That is the loading spinner appears for 3 secs and goes away. This is more of a visual effect than an actual loading state indication. Thinking further I feel we might as well remove this and only show the NProgress bar (there's nowhere as per design files that indicated we should use this spinner; I just did it based on this discussion; https://opencollective.slack.com/archives/C035S573ZD2/p1655212031150739?thread_ts=1655114037.455389&cid=C035S573ZD2). 🤔 Are you okay with that?

SudharakaP replied:
I agree. I removed the spinner from the search form. I think until we find a permanent fix (maybe using NProgress itself) to show a spinner that syncs with NProgress bar we shouldn't add this spinner for the search form or the modal. I've added a commit to remove it.

znarf reviewed (changes requested):
Sorry, that's not working for me.

SudharakaP commented:
> Sorry, that's not working for me.

@znarf : Based on my comment https://github.com/opencollective/opencollective-frontend/pull/7922#discussion_r905038662, I am thinking we shouldn't show this spinner for now until we find a way to either add a spinner that syncs with NProgress bar or use the default spinner by NProgress itself; but currently NProgress doesn't allow their spinner position to be customized (it always appears on the right hand top corner and [we have disabled it](https://github.com/opencollective/opencollective-frontend/blob/f389a2546759893d100042d7f79581ce1482a538/public/static/styles/app.css#L203-L205)).

znarf commented:
I don't understand how we end up removing the spinner completely. Something that we just added based on feedback a few days ago.

I'm closing this.

SudharakaP commented:
> I don't understand how we end up removing the spinner completely. Something that we just added based on feedback a few days ago.
> 
> I'm closing this.

Sorry, I thought that when we added the NProgress bar on the modal that could replace the spinner which is why I removed it due to the consistency issue (until we find a way to sync the spinner with the NProgress bar globally). But if the consistency issue (the spinner is not shown on `/search page) is something we could live with I think we are good to keep it in the search drop-down modal in the homepage. 🤔
